### DISCUSSION TITLE
是否可以不用修改 theme 主题添加 社交媒体配置

### DISCUSSION THREAD

**Discussion by Ayouuuu** [hugo-fixit/FixIt#251] [Q&A] [answered]
如题，如果想要添加自定义社交媒体，需要修改主题下的 `assets/data/social.yml` 文件中的配置，是否可以直接在根目录创建类似的文件添加/覆盖？

**Comment by Lruihao** [comment]
> 如题，如果想要添加自定义社交媒体，需要修改主题下的 `assets/data/social.yml` 文件中的配置，是否可以直接在根目录创建类似的文件添加/覆盖？

可以覆盖该文件，也可以在config.yml中添加

**Reply by Ayouuuu** [reply]
我尝试在根目录创建了 `assets/data/social.yml` 文件，`config.toml` 添加了 `[params.social.Mirror]` 但是他好像并未生效

**Reply by Ayouuuu** [reply]
> > 如题，如果想要添加自定义社交媒体，需要修改主题下的 `assets/data/social.yml` 文件中的配置，是否可以直接在根目录创建类似的文件添加/覆盖？
> 
> 可以覆盖该文件，也可以在config.yml中添加

但是通过修改了 `theme/Fixit/assets/data/social.yml` 文件，再次修改 `config.toml` 可以添加成功

**Reply by Lruihao** [reply]
> 我尝试在根目录创建了 `assets/data/social.yml` 文件，`config.toml` 添加了 `[params.social.Mirror]` 但是他好像并未生效

经测试，该方案是生效的，如下图，social.yml 里的 `Mirror` 你设置小写试试?
<img width="700" alt="image" src="https://user-images.githubusercontent.com/33419593/204436943-5e563b7f-9bac-474e-9bd5-54bca99c842b.png"><img width="200" alt="image" src="https://user-images.githubusercontent.com/33419593/204436977-04990fcb-bc3b-4a9d-a0f0-45c76e4127ed.png"><img width="600" alt="image" src="https://user-images.githubusercontent.com/33419593/204437109-04bda4eb-a58c-478e-9013-bf097caf132b.png">

**Reply by Ayouuuu** [reply]
我检查了页面的元素后发现确实添加了，但是好像并没有读取 `assets/data/social.yml` 中的内容
![image](https://user-images.githubusercontent.com/31991515/204438524-364c7d27-1b06-4708-9622-ca7742091a45.png)
![image](https://user-images.githubusercontent.com/31991515/204438579-8e132f84-36dd-48f5-bc16-a802cd49bb1f.png)

**Reply by Lruihao** [reply]
> 我检查了页面的元素后发现确实添加了，但是好像并没有读取 `assets/data/social.yml` 中的内容
> 
> ![image](https://user-images.githubusercontent.com/31991515/204438524-364c7d27-1b06-4708-9622-ca7742091a45.png)
> 
> ![image](https://user-images.githubusercontent.com/31991515/204438579-8e132f84-36dd-48f5-bc16-a802cd49bb1f.png)
> 
> 

你把主题的 social.yml 复制过去，再额外加新的配置项，修改完，重新启动看看应该没问题的。或者你像我上面那样贴一下你的配置我看看。

**Comment by Ayouuuu** [comment, marked as answer]
破案了，是因为文件名后缀的问题,根目录下只识别 `yml` 后缀，不识别 `yaml` 😭
